engine/robot_logic/traversal.py

```diff
@@ -64,17 +68,34 @@ def move_to_target_node(robot_state, target, allowed_dist_error, database):
     robot_state.goal_location = target
     robot_state.loc_pid_x.reset_integral()
     robot_state.loc_pid_y.reset_integral()
+    using_heading_pid = True
+    simulate = False
+    if simulate:
+        i = 0
+    import matplotlib.pyplot as plt
     while (distance_away > allowed_dist_error) and not (robot_state.phase == Phase.AVOID_OBSTACLE):
         # Error in terms of latitude and longitude, NOT meters
         x_coords_error = target[0] - robot_state.state[0]
         y_coords_error = target[1] - robot_state.state[1]
+        if simulate:
+            i += 1
+            if i == 50:
+                break
+        desired_angle = math.atan2(
+            target[1] - robot_state.state[1], target[0] - robot_state.state[0])
 
         x_vel = robot_state.loc_pid_x.update(x_coords_error)
         y_vel = robot_state.loc_pid_y.update(y_coords_error)
+        if using_heading_pid:
+            turn_to_target_heading(robot_state, desired_angle, 0.01, database)
```

Comment:
unrelated to PIR - currently I know we are doing point turn. However, whenever we decide on turning mechanism, turn_to_target_heading will probably have to be revamped.

Reply:
yep